hip/matrix/bccoo_kernels.hip.cpp

```diff
@@ -0,0 +1,523 @@
+
+2. Redistributions in binary form must reproduce the above copyright
+notice, this list of conditions and the following disclaimer in the
+documentation and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+#include "core/matrix/bccoo_kernels.hpp"
+
+
+#include <hip/hip_runtime.h>
+
+
+#include <ginkgo/core/base/exception_helpers.hpp>
+#include <ginkgo/core/base/math.hpp>
+#include <ginkgo/core/base/types.hpp>
+#include <ginkgo/core/matrix/bccoo.hpp>
+#include <ginkgo/core/matrix/coo.hpp>
+#include <ginkgo/core/matrix/csr.hpp>
+#include <ginkgo/core/matrix/dense.hpp>
+
+
+#include "core/components/fill_array_kernels.hpp"
+#include "core/matrix/bccoo_aux_structs.hpp"
+#include "core/matrix/bccoo_helper.hpp"
+#include "core/matrix/dense_kernels.hpp"
+#include "hip/base/config.hip.hpp"
+#include "hip/base/hipsparse_bindings.hip.hpp"
+#include "hip/base/math.hip.hpp"
+#include "hip/base/types.hip.hpp"
+#include "hip/base/unaligned_access.hip.hpp"
+#include "hip/components/atomic.hip.hpp"
+#include "hip/components/cooperative_groups.hip.hpp"
+#include "hip/components/format_conversion.hip.hpp"
+#include "hip/components/segment_scan.hip.hpp"
+#include "hip/components/thread_ids.hip.hpp"
+
+
+namespace gko {
+namespace kernels {
+/**
+ * @brief The HIP namespace.
+ *
+ * @ingroup hip
+ */
+namespace hip {
+/**
+ * @brief The Bccoordinate matrix format namespace.
+ *
+ * @ingroup bccoo
+ */
+namespace bccoo {
+
+
+using namespace matrix::bccoo;
+
+
+constexpr int default_block_size = 512;
+constexpr int warps_in_block = 4;
+constexpr int spmv_block_size = warps_in_block * config::warp_size;
+
+
+#include "common/cuda_hip/matrix/bccoo_helper.hpp.inc"
+#include "common/cuda_hip/matrix/bccoo_kernels.hpp.inc"
+
+
+template <typename IndexType>
+void get_default_block_size(std::shared_ptr<const HipExecutor> exec,
+                            IndexType* block_size)
+{
+    *block_size = 32;
+}
```

Comment:
It is possible that the default value was different in each executor. 
Now, the easiest solution has been to use the same value, but it could be different.

Reply:
Done!!